File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ChoiceTest/ChoiceTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include <string>
#include "gtest/gtest.h"
#include "Choice.h"
#include "sc_types.h"
Choice* statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new Choice();
		statechart->init();
	}
	virtual void TearDown() {
		delete statechart;
	}
};

void initForEventE(bool valueForC){
	statechart->enter();
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_A));
	statechart->getDefaultSCI()->set_c(valueForC);
	statechart->raise_e();
	statechart->runCycle();
}
void initForEventF(bool valueForC){
	statechart->enter();
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_A));
	statechart->getDefaultSCI()->set_c(valueForC);
	statechart->raise_f();
	statechart->runCycle();
}
void initForEventG(bool valueForC){
	statechart->enter();
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_A));
	statechart->getDefaultSCI()->set_c(valueForC);
	statechart->raise_g();
	statechart->runCycle();
}
void initForEventH(bool valueForC){
	statechart->enter();
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_A));
	statechart->getDefaultSCI()->set_c(valueForC);
	statechart->raise_h();
	statechart->runCycle();
}

TEST_F(StatemachineTest, elseChoiceUsingNonDefaultTransition) {
	
	
	initForEventE(true);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_C));
	
}
TEST_F(StatemachineTest, elseChoiceUsingDefaultTransition) {
	
	
	initForEventE(false);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_B));
	
}
TEST_F(StatemachineTest, defaultChoiceUsingNonDefaultTransition) {
	
	
	initForEventG(true);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_C));
	
}
TEST_F(StatemachineTest, defaultChoiceUsingDefaultTransition) {
	
	
	initForEventG(false);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_B));
	
}
TEST_F(StatemachineTest, uncheckedChoiceUsingNonDefaultTransition) {
	
	
	initForEventF(true);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_C));
	
}
TEST_F(StatemachineTest, uncheckedChoiceUsingDefaultTransition) {
	
	
	initForEventF(false);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_B));
	
}
TEST_F(StatemachineTest, alwaysTrueTransitionInChoice) {
	
	
	initForEventH(true);
	
	EXPECT_TRUE(statechart->isStateActive(Choice::main_region_C));
	
}

File: test-plugins/org.yakindu.sct.generator.java.test/test-gen/org/yakindu/sct/generator/java/test/ChoiceTest.java
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/

package org.yakindu.sct.generator.java.test;

import org.junit.*;
import static org.junit.Assert.*;
import org.yakindu.scr.choice.ChoiceStatemachine;
import org.yakindu.scr.choice.ChoiceStatemachine.State;

/**
 * Unit TestCase for Choice
 */
@SuppressWarnings("all")
public class ChoiceTest {
	private ChoiceStatemachine statemachine;	
	
	@Before
	public void setUp() {
		statemachine = new ChoiceStatemachine();
		statemachine.init();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}
	
	@Test
	public void testElseChoiceUsingNonDefaultTransition() {
		initForEventE(true);
		assertTrue(statemachine.isStateActive(State.main_region_C));
	}
	@Test
	public void testElseChoiceUsingDefaultTransition() {
		initForEventE(false);
		assertTrue(statemachine.isStateActive(State.main_region_B));
	}
	@Test
	public void testDefaultChoiceUsingNonDefaultTransition() {
		initForEventG(true);
		assertTrue(statemachine.isStateActive(State.main_region_C));
	}
	@Test
	public void testDefaultChoiceUsingDefaultTransition() {
		initForEventG(false);
		assertTrue(statemachine.isStateActive(State.main_region_B));
	}
	@Test
	public void testUncheckedChoiceUsingNonDefaultTransition() {
		initForEventF(true);
		assertTrue(statemachine.isStateActive(State.main_region_C));
	}
	@Test
	public void testUncheckedChoiceUsingDefaultTransition() {
		initForEventF(false);
		assertTrue(statemachine.isStateActive(State.main_region_B));
	}
	@Test
	public void testAlwaysTrueTransitionInChoice() {
		initForEventH(true);
		assertTrue(statemachine.isStateActive(State.main_region_C));
	}
	public void initForEventE(boolean valueForC) {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_A));
		statemachine.setC(valueForC);
		statemachine.raiseE();
		statemachine.runCycle();
	}
	public void initForEventF(boolean valueForC) {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_A));
		statemachine.setC(valueForC);
		statemachine.raiseF();
		statemachine.runCycle();
	}
	public void initForEventG(boolean valueForC) {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_A));
		statemachine.setC(valueForC);
		statemachine.raiseG();
		statemachine.runCycle();
	}
	public void initForEventH(boolean valueForC) {
		statemachine.enter();
		assertTrue(statemachine.isStateActive(State.main_region_A));
		statemachine.setC(valueForC);
		statemachine.raiseH();
		statemachine.runCycle();
	}
}
